specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyEvents_QueryResourceScopeExpandComponentsGroupByWithAggregate.json

```diff
@@ -0,0 +1,59 @@
+    "$expand": "components($apply=groupby((tenantId, principalOid, policyDefinitionAction), aggregate($count as totalActions)))"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "@odata.context": "https://management.azure.com/subscriptions/e78961ba-36fe-4739-9212-e3031b4c8db7/resourcegroups/myResourceGroup/providers/microsoft.keyvault/vaults/myKVName/providers/Microsoft.PolicyInsights/policyEvents/$metadata#default",
+        "@odata.count": 1,
+        "value": [
+          {
+            "@odata.id": null,
+            "@odata.context": "https://management.azure.com/subscriptions/e78961ba-36fe-4739-9212-e3031b4c8db7/resourcegroups/myResourceGroup/providers/microsoft.keyvault/vaults/myKVName/providers/Microsoft.PolicyInsights/policyEvents/$metadata#default/$entity",
+            "timestamp": "2020-02-21T19:42:08.3252921Z",
+            "resourceId": "/subscriptions/e78961ba-36fe-4739-9212-e3031b4c8db7/resourcegroups/myResourceGroup/providers/microsoft.keyvault/vaults/myKVName",
+            "policyAssignmentId": "/subscriptions/e78961ba-36fe-4739-9212-e3031b4c8db7/providers/microsoft.authorization/policyassignments/560050f83dbb4a24974323f8",
+            "policyDefinitionId": "/subscriptions/e78961ba-36fe-4739-9212-e3031b4c8db7/providers/microsoft.authorization/policydefinitions/ab108bc4-32df-4677-8b38-fa8b2905df59",
+            "effectiveParameters": "",
+            "isCompliant": false,
+            "subscriptionId": "e78961ba-36fe-4739-9212-e3031b4c8db7",
+            "resourceType": "Microsoft.KeyVault/vaults",
+            "resourceLocation": "westcentralus",
+            "resourceGroup": "myResourceGroup",
+            "resourceTags": "tbd",
+            "policyAssignmentName": "560050f83dbb4a24974323f8",
+            "policyAssignmentOwner": "tbd",
+            "policyAssignmentParameters": "",
+            "policyAssignmentScope": "/subscriptions/e78961ba-36fe-4739-9212-e3031b4c8db7",
+            "policyDefinitionName": "ab108bc4-32df-4677-8b38-fa8b2905df59",
+            "policyDefinitionAction": "audit",
+            "policyDefinitionCategory": "tbd",
+            "policySetDefinitionId": "",
+            "policySetDefinitionName": "",
+            "policySetDefinitionOwner": "",
+            "policySetDefinitionCategory": "",
+            "policySetDefinitionParameters": "",
+            "managementGroupIds": "PolicyUIMG,AzGovTest5,72f988bf-86f1-41af-91ab-2d7cd011db47",
+            "policyDefinitionReferenceId": "",
+            "complianceState": "NonCompliant",
+            "tenantId": "",
+            "principalOid": "",
+            "components": [
+                {
+                    "tenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
+                    "principalOid": "0d81b461-6bb0-4909-a102-d51803a7d275",
+                    "policyDefinitionAction": "audit",
+                    "totalActions": 6
```

Comment:
Property of `totalActions` is not defined in the swagger [here](https://github.com/Azure/azure-rest-api-specs/blob/77c82d458cdef45277b91b4351075383e0eee776/specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json#L755)

Reply:
The issue must be something else, because there are quite a few other json files that are written in the same manner, where they take custom property names from the  request URL and are not defined in the schema, for some reason model validation is passing for these root properties but not for nested properties(that are in my PR). Here is an example of similar json for which validation is passing:
https://github.com/Azure/azure-rest-api-specs/blob/77c82d458cdef45277b91b4351075383e0eee776/specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyEvents_FilterAndAggregateOnly.json

If you look in this example NumDenyEvents property is not defined in the swagger for this API, but validation is passing.